st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -441,28 +444,30 @@ def _consume_stderr(self, chan):
         Try to consume stderr data from chan if it's receive ready.
         """
 
+        out = bytearray()
         stderr = StringIO()
         if chan.recv_stderr_ready():
             data = chan.recv_stderr(self.CHUNK_SIZE)
+            out += data
 
             while data:
-                stderr.write(self._get_decoded_data(data))
                 ready = chan.recv_stderr_ready()
 
                 if not ready:
                     break
 
                 data = chan.recv_stderr(self.CHUNK_SIZE)
+                out += data
 
+        stderr.write(self._get_decoded_data(out))
```

Comment:
(thinking out loud)

Since we only write to StringIO object once now the only reason to keep it is for the API to be backward compatible.